test/chain-migration-test.js

```diff
@@ -664,6 +664,121 @@ describe('Chain Migrations', function() {
         'Chain State did not properly migrate.');
     });
   });
+
+  describe('Migration Blockstore (integration)', function() {
+    const location = testdir('migrate-chain-state');
+    const migrationsBAK = ChainMigrator.migrations;
+    const store = BlockStore.create({
+      memory: false,
+      prefix: location,
+      network
+    });
+
+    const workers = new WorkerPool({
+      enabled: true,
+      size: 2
+    });
+
+    const chainOptions = {
+      prefix: location,
+      memory: false,
+      blocks: store,
+      network,
+      workers
+    };
+
+    let chain, miner, cpu;
+    before(async () => {
+      ChainMigrator.migrations = {};
+      await fs.mkdirp(location);
+      await store.ensure();
+      await workers.open();
+    });
+
+    after(async () => {
+      ChainMigrator.migrations = migrationsBAK;
+      await rimraf(location);
+      await workers.close();
+    });
+
+    beforeEach(async () => {
+      chain = new Chain(chainOptions);
+      miner = new Miner({ chain });
+      cpu = miner.cpu;
+
+      await miner.open();
+      await store.open();
+      await chain.open();
+    });
+
+    afterEach(async () => {
+      await chain.close();
+      await store.close();
+      await miner.close();
+    });
+
+    const blocks = [];
+    it('should mine 10 blocks', async () => {
+      for (let i = 0; i < 10; i++) {
+        const block = await cpu.mineBlock();
+        assert(block);
+        assert(await chain.add(block));
+        blocks.push(block);
+      }
+    });
+
+    it('should move blocks back to ldb.', async () => {
```

Comment:
The content itself does not really make a difference. It could be just random bytes (as you can observe in unit tests for the blockstore itself). This test is here to make sure everything around blockstore behaves properly - integration test for chain + blockstore.